overlord/hookstate/ctlcmd/mount.go

```diff
@@ -0,0 +1,223 @@
+)
+
+var (
+	shortMountHelp = i18n.G("Create a temporary or permanent mount")
+	longMountHelp  = i18n.G(`
+The mount command mounts the given source onto the given destination path.`)
+)
+
+func init() {
+	addCommand("mount", shortMountHelp, longMountHelp, func() command { return &mountCommand{} })
+}
+
+type mountCommand struct {
+	baseCommand
+	Positional struct {
+		What  string `positional-arg-name:"<what>" required:"yes" description:"path to the resource to be mounted"`
+		Where string `positional-arg-name:"<where>" required:"yes" description:"path to the destination mount point"`
+	} `positional-args:"yes" required:"yes"`
+	Persistent  bool   `long:"persistent" description:"make the mount persist across reboots"`
+	Type        string `long:"type" short:"t" description:"partition type"`
+	Options     string `long:"options" short:"o" description:"comma-separated list of mount options"`
+	snapInfo    *snap.Info
+	optionsList []string
+}
+
+func matchMountPathAttribute(path string, attribute interface{}, snapInfo *snap.Info) bool {
+	pattern, ok := attribute.(string)
+	if !ok {
+		return false
+	}
+
+	expandedPattern := snapInfo.ExpandSnapVariables(pattern)
+
+	pp, err := utils.NewPathPattern(expandedPattern)
+	return err == nil && pp.Matches(path)
+}
+
+// matchConnection checks whether the given mount connection attributes give
+// the snap permission to execute the mount command
+func (m *mountCommand) matchConnection(attributes map[string]interface{}) bool {
+	if !matchMountPathAttribute(m.Positional.What, attributes["what"], m.snapInfo) {
+		return false
+	}
+
+	if !matchMountPathAttribute(m.Positional.Where, attributes["where"], m.snapInfo) {
+		return false
+	}
+
+	if m.Type != "" {
+		// TODO: what should we do if the type is not given?
+		if types, ok := attributes["type"].([]interface{}); ok {
+			found := false
+			for _, iface := range types {
+				if typeString, ok := iface.(string); ok && typeString == m.Type {
+					found = true
+					break
+				}
+			}
+			if !found {
+				return false
+			}
+		} else {
+			return false
+		}
+	}
+
+	if optionsIfaces, ok := attributes["options"].([]interface{}); ok {
+		var allowedOptions []string
+		for _, iface := range optionsIfaces {
+			if option, ok := iface.(string); ok {
+				allowedOptions = append(allowedOptions, option)
+			}
+		}
+		for _, option := range m.optionsList {
+			if !strutil.ListContains(allowedOptions, option) {
+				return false
+			}
+		}
+	}
+
+	if m.Persistent {
+		if allowedPersistent, ok := attributes["persistent"].(bool); !ok || !allowedPersistent {
+			return false
+		}
+	}
+
+	return true
+}
+
+// checkConnections checks whether the established connections give the snap
+// permission to execute the mount command
+func (m *mountCommand) checkConnections(context *hookstate.Context) error {
+	snapName := context.InstanceName()
+
+	st := context.State()
+	st.Lock()
+	defer st.Unlock()
+
+	conns, err := ifacestate.ConnectionStates(st)
```

Comment:
for a follow up but now I think we have 2 or 3 places in ctlcmd doing matches against this, maybe they can share a helper? worth checking at least